python/common/org/python/types/Str.java

```diff
@@ -860,9 +860,18 @@ private long toPositiveIndex(long index) {
                     "False otherwise.\n"
     )
     public org.python.Object isnumeric() {
-        throw new org.python.exceptions.NotImplementedError("isnumeric() has not been implemented.");
+        if (this.value.isEmpty()) {
+            return new org.python.types.Bool(false);
+        }
+        for (char ch : this.value.toCharArray()) {
+            if (!(Character.isDigit(ch))) {
+                return new org.python.types.Bool(false);
+            }
+        }
+        return new org.python.types.Bool(true);
     }
 
+
```

Comment:
I believe there's just one newline between the other methods in this class – so I think it's not necessary to add this newline here.